### PR TITLE
[API] Browsing payments

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | api
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingPaymentsContext.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    public function thePaymentOfTheOrderShouldBeFor(
+        string $orderNumber,
+        string $paymentState,
+        CustomerInterface $customer
+    ): void {
+        foreach ($this->client->getCollectionItemsWithValue('state', StringInflector::nameToLowercaseCode($paymentState)) as $payment) {
+            $this->client->showByIri($payment['order']);
+            if ($this->client->responseHasValue('number', $orderNumber)) {
+                $this->client->showByIri($this->client->getResponseValue('customer'));
+                if ($this->client->responseHasValue('email', $customer->getEmail())) {
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Admin/ManagingPaymentsContext.php]: This amount of indention is sad, but I guess would be fine for the first iteration over this

Comment (GSadee) [reply src/Sylius/Behat/Context/Api/Admin/ManagingPaymentsContext.php]: I've done the first minor refactor of this step 👌

**Comment by Zales0123** [comment]
Thank you, Grzegorz! :1st_place_medal:

File: src/Sylius/Behat/Client/ApiPlatformClient.php [hunk 2]
```diff
@@ -153,6 +163,14 @@ public function responseHasValue(string $key, $value): bool
         return $this->getResponseContentValue($key) === $value;
     }
 
+    /** @param string|int $value */
+    public function relatedResourceHasValue(string $resource, string $key, $value): bool
+    {
+        $this->showRelated($resource);
+
+        return $this->getResponseContentValue($key) === $value;
+    }
```
Comment (Tomanhez) [inline src/Sylius/Behat/Client/ApiPlatformClient.php]: Wrong order , I guess this method should be above `public function responseHasValue(string $key, $value): bool`

**Comment by lchrusciel** [comment]
Part of #11250

**Pull request merged**